runners/spark/src/main/java/org/apache/beam/runners/spark/aggregators/NamedAggregators.java

```diff
@@ -39,170 +41,185 @@
  * is launched. We can then add aggregators on the fly in Spark.
```

Comment:
@bjchambers I might be able to shed some light here. 
Spark "Aggregators" are called `Accumulators` (see [here](http://spark.apache.org/docs/1.6.2/programming-guide.html#accumulators) for details), and Spark requires the Accumulator be created outside of the applied function (which is the equivalent of a `DoFn`) and be passed to it, sort of like a distributed counter that is pre-defined for all DoFns.

Beam Aggregators on the other hand, are defined within and by the `DoFn` - one example of the difference is being able to create an Aggregator corresponding to the processed data.

The way the Spark runner bridges this gap is by defining one, uber-Accumulator,  that is actually a map of Aggregators (or to be more accurate a map between the Aggregator name and it's state, see [here](https://github.com/apache/incubator-beam/blob/master/runners/spark/src/main/java/org/apache/beam/runners/spark/aggregators/NamedAggregators.java)).

Finally, Spark uses dropwizard metrics internally (not with Accumulators), and some people in the Spark community have used [this](https://gist.github.com/ibuenros/9b94736c2bad2f4b8e23#file-sparkutils-scala-L25) "instrumentation" to add the user-defined Accumulators to the metrics `Reporter` thus unifying metrics and Accumulators and being able to report Accumulator values to the metrics system as well.

As for this PR, it should take care of the fact that the SparkRunner is not aware of any Aggregator until it produces values so it can't be registered by the mentioned "instrumentation" once and ahead of time, so instead it should update the registered metrics continuously.

Hope this helps!